File: 539. Minimum Time Difference/findMinDifference.py
from typing import List

class Solution:
  def transformToMinutes(self, timePoints: List[str]) -> List[int]:
    minutes = []
    for time in timePoints:
      hour, minute = time.split(':')
      minutes.append(int(hour) * 60 + int(minute))

    return minutes

  def findMinDifference(self, timePoints: List[str]) -> int:
    minutes = self.transformToMinutes(timePoints)
    minutes.sort()
    minDifference = 24 * 60

    for mins in range(1, len(minutes)):
      minDifference = min(minDifference, minutes[mins] - minutes[mins - 1])

    firstLastDifference = minutes[0] + 24 * 60 - minutes[-1]
    minDifference = min(minDifference, firstLastDifference)

    return minDifference


timePoints = ["23:59","00:00"]
solution = Solution()
print("findMinDifference: ", solution.findMinDifference(timePoints)) # 1

File: 539. Minimum Time Difference/findMinDifference.js
/**
 * @param {string[]} timePoints
 * @return {number}
 */
const findMinDifference = (timePoints) => {
  const minutes = timePoints.map(item => {
    const [hours, minutes] = item.split(':');
    return parseInt(hours) * 60 + parseInt(minutes);
  });

  minutes.sort((a, b) => a - b);
  let minDiff = 24 * 60;

  for (let i = 1; i < minutes.length; i++) {
    minDiff = Math.min(minDiff, minutes[i] - minutes[i - 1]);
  }

  const firstLastDiff = minutes[0] + 24 * 60 - minutes[minutes.length - 1];
  minDiff = Math.min(minDiff, firstLastDiff);

  return minDiff;
};


const timePoints = ["23:59", "00:00"];
console.log("findMinDifference: ", findMinDifference(timePoints)); // 1
